Azure/azure-cli
# [Extension] Add azmirror cloud endpoint for installing extensions in air gapped clouds and support config of index url

**Description**  

To support installing extensions in air-gapped clouds, we have set up a [pipeline](https://dev.azure.com/azure-sdk/internal/_build?definitionId=1608) to sync the extension index, wheel packages and extension command tree to a public storage account called `azmirror`. A Teleport process will be set up to sync everything in `azmirror` to different storage accounts inside air-gapped clouds. 

In order to use those files in each air-gapped cloud with Azure CLI, this PR adds an `azmirror_storage_account_resource_id` endpoint whose value may look like `https://azmirroragc.blob.core.windows.net`.
CLI will get extension related files from the endpoint if it is set in the current cloud, otherwise it will use the current default values (for public cloud and sovereign clouds).

This PR also adds config option `extension.index_url` to support customized extension index url. For example, to use the `index.json` on github:
`az config set extension.index_url=https://raw.githubusercontent.com/Azure/azure-cli-extensions/master/src/index.json`

**Testing Guide**  

```
$ az cloud update --cloud-config "{\"endpoint_azmirror_storage_account_resource_id\":\"https://azmirror.blob.core.windows.net\"}"
$ az cloud show  # you should see the value set for azmirrorStorageAccountResourceId
$ az extension list-available -otable --debug  # you should see a request sent to https://azmirror.blob.core.windows.net
$ az config set extension.index_url=https://raw.githubusercontent.com/Azure/azure-cli-extensions/master/src/index.json
$ az extension list-available -otable --debug  # Now you should see a request sent to raw.githubusercontent.com
```
**History Notes**  


[Extension] Support config of extension index url

---

This checklist is used to make sure that common guidelines for a pull request are followed.
